src/Foundatio/Utility/SystemClock.cs

```diff
@@ -1,175 +1,118 @@
 
-        public DateTime Now() {
-            return DateTime.Now;
+        protected SchedulerSystemClockBase(IScheduler scheduler)
+        {
+            _scheduler = scheduler;
         }
 
-        public DateTime UtcNow() {
-            return DateTime.UtcNow;
+        public IDisposable Schedule<TState>(TState state, Func<IScheduler, TState, IDisposable> action)
+        {
+            return _scheduler.Schedule(state, action);
         }
 
-        public DateTimeOffset OffsetNow() {
-            return DateTimeOffset.Now;
+        public IDisposable Schedule<TState>(TState state, TimeSpan dueTime, Func<IScheduler, TState, IDisposable> action)
+        {
+            return _scheduler.Schedule(state, dueTime, action);
         }
 
-        public DateTimeOffset OffsetUtcNow() {
-            return DateTimeOffset.UtcNow;
+        public IDisposable Schedule<TState>(TState state, DateTimeOffset dueTime, Func<IScheduler, TState, IDisposable> action)
+        {
+            return _scheduler.Schedule(state, dueTime, action);
         }
 
-        public void Sleep(int milliseconds) {
-            Thread.Sleep(milliseconds);
-        }
+        public abstract CancellationTokenSource CreateCancellationTokenSource(TimeSpan timeout);
 
-        public Task SleepAsync(int milliseconds, CancellationToken ct) {
-            return Task.Delay(milliseconds, ct);
-        }
-
-        public TimeSpan TimeZoneOffset() {
-            return DateTimeOffset.Now.Offset;
-        }
+        public DateTimeOffset Now => _scheduler.Now;
     }
 
-    public class TestSystemClock : ISystemClock {
-        private DateTime? _fixedUtc = null;
-        private TimeSpan _offset = TimeSpan.Zero;
-        private TimeSpan _timeZoneOffset = DateTimeOffset.Now.Offset;
-        private bool _fakeSleep = false;
-
-        public DateTime UtcNow() {
-            return _fixedUtc ?? DateTime.UtcNow.Add(_offset);
-        }
-
-        public DateTime Now() {
-            return UtcNow().Add(_timeZoneOffset);
-        }
+    public sealed class DefaultSystemClock : SchedulerSystemClockBase {
+        public static readonly DefaultSystemClock Instance = new DefaultSystemClock();
 
-        public DateTimeOffset OffsetNow() {
-            return new DateTimeOffset(Now().Ticks, _timeZoneOffset);
+        public DefaultSystemClock() : base(DefaultScheduler.Instance)
+        {
         }
 
-        public DateTimeOffset OffsetUtcNow() {
-            return new DateTimeOffset(UtcNow().Ticks, TimeSpan.Zero);
+        public override CancellationTokenSource CreateCancellationTokenSource(TimeSpan timeout)
+        {
+            return new CancellationTokenSource(timeout);
         }
+    }
 
-        public void Sleep(int milliseconds) {
-            if (!_fakeSleep) {
-                Thread.Sleep(milliseconds);
-                return;
+    public static class SystemClock {
+        private static readonly AsyncLocal<ISystemClock> _instance = new AsyncLocal<ISystemClock>();
```

Comment:
I'm not really sure what this is accomplishing. The class and instance are static. So it's not like the instance is scoped to only callers inside of the using blocks.